### PR TITLE
feat(protocol-designer): wire up rename step

### PR DESCRIPTION
**Pull request by syao1226**
re AUTH-805

# Overview

Update the rename step feature to allow users to rename a step and add step notes when they click the rename button on step forms.

## Test Plan and Hands on Testing

- Go to Protocol steps and add a step
- Click the rename button and confirm that a modal appears
- Save the new step name and step notes
- Verify that the step name is renamed (the step details will be added below the command summary as the next task).
- Ensure it matches the [design](https://www.figma.com/design/WbkiUyU8VhtKz0JSuIFA45/Feature%3A-Protocol-Designer-Phase-1?node-id=11219-264190&t=1OIUHns25vPqdMDK-4)

## Changelog

- Created a new `RenameStepModal` under the `organisms` directory to display a modal for users to update the step name and step details
- Added `renameStep()` and `RenameStepAction` interface in `labware-ingred/actions/actions.ts` to handle changes
- Updated `StepFormToolbox` to handle rename button click
- Added a unit test for `RenameStepModal`
## Review requests

- I used `CHANGE_FORM_INPUT` as the type for the `RenameStepAction` interface, taking it from the old PD `ChangeFormInputAction` to handle `stepName` and `stepDetails` updates. Am I supposed to use this, or should I create a new type, add it to `labware-ingred/actions/actions.ts`, and use that instead?

### CODE REVIEW

File: protocol-designer/src/labware-ingred/actions/actions.ts [hunk 1]
```diff
@@ -295,3 +297,18 @@ export const generateNewProtocol: (
...
+    update: Partial<Record<StepFieldName, unknown | null>>
+  }
+}
+
+export const renameStep: (
+  payload: RenameStepAction['payload']
+) => RenameStepAction = payload => ({
+  type: 'CHANGE_FORM_INPUT',
+  payload,
+})
```
Comment (jerader) [inline protocol-designer/src/labware-ingred/actions/actions.ts]: nice, while this does work, i do have some concerns that we have another action also of type `CHANGE_FORM_INPUT`. If you don't want to use the existing `changeFormInput` action, i recommend calling this one something else (something like `CHANGE_STEP_DETAILS` or something). Then you'd also have to add it to the `unsavedForm` reducer in `protocol-designer/src/step-forms/reducers/index.ts`

**Review by jerader — COMMENTED** [review]
the functionality works as expected, nice work! just left a comment regarding the new action you made.

File: protocol-designer/src/organisms/RenameStepModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+const MAX_STEP_NAME_LENGTH = 60
+interface RenameStepModalProps {
+  formData: FormData
+  onClose: () => void
+}
+export function RenameStepModal(props: RenameStepModalProps): JSX.Element {
+  const { onClose, formData } = props
+  const dispatch = useDispatch()
+  const { t } = useTranslation(['form', 'shared', 'protocol_steps'])
```
Comment (koji) [inline protocol-designer/src/organisms/RenameStepModal/index.tsx:33]:
```suggestion
  const { t } = useTranslation(['form', 'shared', 'protocol_steps'])
  const dispatch = useDispatch()
```

**Review by jerader — APPROVED** [review]
lgtm. nice work making a new Redux action & adding onto the reducer!

**Pull request merged**